src/api/index.js

```diff
@@ -1,56 +1,49 @@
-import RollAuthentication from "./authentication";
-import RollUserAPI from "./user";
-import RollRequest from "./request";
-import RollTransactionAPI from "./transaction";
+import { auth } from "@tryrolljs/sdk";
+import { client } from "@tryrolljs/api";
 import config from "../config";
 
-class RollAPI {
-  constructor({ apiURL, oauthConfig, authCacheConfig }) {
-    this.authentication = new RollAuthentication(oauthConfig, authCacheConfig);
-    const request = new RollRequest(this.authentication.getAuthToken, apiURL);
-    this.user = new RollUserAPI(this.authentication.getAuthToken, request);
-    this.transaction = new RollTransactionAPI(
-      this.authentication.getAuthToken,
-      request
-    );
-  }
-}
-
-const OAUTH_KEY = "oauthTokens";
-
-// Implement custom functions to cache tokens between sessions
-const cacheOauthTokens = (oauthTokens) => {
-  localStorage.setItem(OAUTH_KEY, oauthTokens);
+const authSdkConfig = {
+  clientId: config.clientID,
+  issuerUrl: config.issuerURL,
+  redirectUrl: config.redirectURL,
+  logoutRedirectUrl: config.redirectURL,
+  scopes: config.scopes,
 };
 
-const getCachedOauthTokens = () => {
-  return localStorage.getItem(OAUTH_KEY);
-};
+const authSdk = new auth.AuthSDK(authSdkConfig, localStorage);
 
-const clearCachedOauthTokens = () => {
-  localStorage.removeItem(OAUTH_KEY);
+const apiClientConfig = {
+  getApiUrl: () => config.apiURL,
+  getAuthorization: () => {
+    const accessToken = authSdk.getAccessToken();
+    return accessToken ? `Bearer ${accessToken}` : undefined;
+  },
+  getAuthorizationExpired: authSdk.isTokenExpired,
 };
 
-// update the oauth configuration to log in with a different client
-const oauthConfig = {
-  clientID: config.clientID,
-  issuerURL: config.issuerURL,
-  redirectURL: config.redirectURL,
-  scopes: config.scopes,
+const apiClientParsers = {
+  error: (response) => {
+    if (typeof response.data === "object") {
```

Comment:
this seems like something that should be part of the api package. This should be the default behavior, which the consumer can hook into if they want to.

Reply:
✅